### PR TITLE
[Enhancement] Adding a --version switch

### PR DESCRIPTION
**Pull request by majorthorn**
### **Suggestion:**
I suggest adding a --version switch to the script to allow for the installation of versions that are not the same as the hardcoded version number on the script. This way if someone wants to install a newer/older version they don't have to wait on the script to be updated to show the new version or edit the scrip themselves.

### **Changes:**

Added --version to help

Added --version switch for hardcoded version override.

Added if statement to check for --version
If there is nothing provided after version the script will continue with the hardcoded version.

Moved package name down to allow for the new version switch to change the package name.

### CODE REVIEW

**Comment by canha** [comment]
This looks good, but could we had some tests to cover this specific case? As in, we put in a version and the expected output is the version that we initially put in.

**Comment by majorthorn** [comment]
I think i understand what you are asking, but I am new to shell scripting. Can you elaborate on what you mean by tests?

**Comment by canha** [comment]
I'm using Travis to run a few shell scripts for us that test if the script works as expected, it's pretty much very basic stuff. You can see how it runs under the [.travis.yml](https://github.com/canha/golang-tools-install-script/blob/master/.travis.yml) file. Perhaps we can add a specific bash test there (inside our tests folder) to make test this specific case (with the --version flag)?

**Comment by majorthorn** [comment]
CI's are a new concept to me. I haven't used them in the past. Would it be just as easy as making a script like version.sh that runs the script with something like ./goinstall.sh --version 1.12 ??

**Comment by majorthorn** [comment]
@canha  I think i am understanding. I just added Travis CI to my fork and I am running with my theory.

**Comment by canha** [comment]
Yes. Feel free to experiment, Having a specific bash file to test this seems fine by me. You'll need to do some magic to verify that the output of `go --version` matches the version you specified but something like `grep` can help there.

**Comment by majorthorn** [comment]
i am hitting roadblocks on it when i pass the --version 1.12 to the CI it is coming back as file not found. so testing this may take a lot more trial and error.

**Comment by majorthorn** [comment]
I guess we can keep this PR on hold until I figure this out. Gives me a chance to learn CI

Thanks

**Comment by majorthorn** [comment]
Alright, I think I finally got the tests working the way you wanted it to work. Let me know what you think.

File: goinstall.sh [hunk 1]
```diff
@@ -66,6 +65,13 @@ if [ "$1" == "--remove" ]; then
 elif [ "$1" == "--help" ]; then
     print_help
     exit 0
+elif [ "$1" == "--version" ]; then
+    if [ -z "$2" ]; then # Check if --version has a second positional parameter
+        echo "Please provide a version number for: $1"
+        
+    else
+        VERSION=$2
+fi
```
Comment (canha) [inline goinstall.sh]: Can you fix the indentation here please?

File: goinstall.sh [hunk 2]
```diff
@@ -66,6 +65,13 @@ if [ "$1" == "--remove" ]; then
 elif [ "$1" == "--help" ]; then
     print_help
     exit 0
+elif [ "$1" == "--version" ]; then
+    if [ -z "$2" ]; then # Check if --version has a second positional parameter
+        echo "Please provide a version number for: $1"
+
```
Comment (canha) [inline goinstall.sh]: Don't need this extra line

File: goinstall.sh (hunk 1 above)
Comment (majorthorn) [reply goinstall.sh]: The indentation of Version=$2?

File: goinstall.sh (hunk 2 above)
Comment (majorthorn) [reply goinstall.sh]: done

File: goinstall.sh (hunk 1 above)
Comment (canha) [reply goinstall.sh]: For the fi :) It's off

Comment (majorthorn) [reply goinstall.sh]: Ah! that is because it looks like we are missing a fi since it is a nested If statement

Comment (majorthorn) [reply goinstall.sh]: @canha Fixed with https://github.com/canha/golang-tools-install-script/pull/25/commits/9a61b03a8dd41543955980083f1f8370e5564616

**Comment by canha** [comment]
Looks good, cheers!

**Pull request merged**